### PR TITLE
Backport Support multiple ext_authz filters (enterprise-only)

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Work in Progress
Not a protected branch, so leaving here until I get approval

# Description

Backport of: https://github.com/solo-io/gloo/pull/4870

Introduce the API for a new enterprise-only feature, where multiple auth servers can be configured in settings, and opted into on a per-route basis.

# Context

Customers have different SLAs for routes/services and want the ability to delegate to separate external auth services (with separate configurations). Expose the ability to configure multiple ext_authz filters (https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto.html) and allow enterprise users to opt into 1 of these extauth services at a time.

# Why is this PR so large?
To release this feature, we need to backport a number of other changes.

### https://github.com/solo-io/gloo/issues/4020
This feature (supporting multiple ext_authz filters) relies on plugins in gloo open source and gloo enterprise being upgradable. That means that instead of both being executed during the translation loop, only the upgraded (enterprise) plugin is evaluated. However, the upgradable plugin feature was introduced in a `v1.7.x beta` and was not available in `1.6.x`.

I opted to backport the upgradable feature, since this is a one-off stub, and because it keeps the extauth plugin functionality clean and in sync with what we have on master. The upgradable feature was introduced here: https://github.com/solo-io/gloo/pull/4052. As you can see, that PR is large since it introduces placeholder plugins for any enterprise features.

### https://github.com/solo-io/gloo/issues/4555
Fix enterprise install commands to use latest patch version based off of the CLI minor version.

### https://github.com/solo-io/gloo/pull/4365
Cleanup our docs gen github workflow to ensure that docs are only build on pull requests to master. Consolidate the logic into a single place so that we maintain a single source of truth.

### https://github.com/solo-io/gloo/pull/4870
I then cherry-picked the feature I wanted to introduce (https://github.com/solo-io/gloo/pull/4870/) and updated first the protos, then the plugins and syncer packages, and finally the generated code.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sam-heilbron** [comment]
/skip-changelog bot doesn't understand 1.6.8-patch1 to be a greater version than 1.6.8

File: projects/gloo/cli/pkg/cmd/install/install_test.go [hunk 1]
```diff
@@ -133,6 +133,8 @@ var _ = Describe("Install", func() {
 	})
 
 	It("should not error when providing the admin console flag", func() {
+		version.Version = "1.3.2" // pretend we set this using linker on a release build of glooctl
```
Comment (sam-heilbron) [inline projects/gloo/cli/pkg/cmd/install/install_test.go:136]: This is what we did in the 1.6.x branch (https://github.com/solo-io/gloo/pull/4548/files#diff-65d626e8d0ed9d2561fe241c94412eda9e65bed8ee967bc1b2a51c2acc6a04dfR134)

**Comment by sam-heilbron** [comment]
/kick ec2 test flake https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=4b95c596-4030-453f-8494-679b6558d818

**Review by kdorosh — COMMENTED** [review]
lgtm

**Pull request merged**